presto-iceberg/src/main/java/com/facebook/presto/iceberg/PrestoIcebergNestedField.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.iceberg;
+
+import com.facebook.presto.common.type.Type;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrestoIcebergNestedField
+{
+    private final String isOptional;
+    private final int id;
+    private final String name;
+    private final Type prestoType;
+    private final String doc;
+
+    @JsonCreator
+    public PrestoIcebergNestedField(
+            @JsonProperty("isOptional") String isOptional,
```

Comment:
why is this a string and not boolean?

Reply:
Yes, this was temporary, I was trying out something. For some reason, when taskUpdateRequest is getting serialized on the coordinator and then deserialized on the worker, the true boolean value becomes false after deserialization. 

To see why that is happening:

https://github.com/prestodb/presto/blob/c254d194e5151d1b01ee276001372f0f229191f1/presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java#L859

This will be part of updateRequest.writeInfo. Before this object is serialized it has the correct value of true for `isOptional` but after the serialization, if I am trying to deserialize it using `taskUpdateRequestCodec.fromBytes(taskUpdateRequestJson)` the object returned has a value of false for `isOptional`. 

Now, when I rename the field `isOptional` to just `optional`, it can serialize and deserialize correctly. 

I will try to understand how the serialization/deserialization is affected by the name of a boolean field. In case you have an idea, please let me know. 

In any case, I will rename the field in this PR.

Reply:
I think renaming to avoid the `is` collision should be fine. I think you are seeing serialization issues because the json property here is defined as `isOptional` but your `@JsonProperty` method for the field is `isOptional`, where it probably should be `getIsOptional` or `isIsOptional` (I don't know which is correct). Another option is to keep the method name and change the `@JsonProperty` annotation to be `@JsonProperty("isOptional")`.

I think it's just best to avoid `is` in the property name anyways to prevent future confusion.

Reply:
Yes, since this was a wrapper class for one of the iceberg library class NestedField, I was trying to keep the field names the same, and the field name in the library was isOptional. As you pointed out, I think the issue would be the getter method's name due to which it is not able to pick the correct value. 

I have anyways renamed the field to `optional` now since that looks better.